platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/DefaultComponentResolversFactory.java

```diff
@@ -127,4 +137,56 @@ private ComponentResolvers createModuleRepositoryResolvers(
         );
     }
 
+    /**
+     * Determines if the repository should not be used to resolve the requested resolve context.
+     */
+    private static boolean shouldSkipRepository(
```

Comment:
💭 This method could just accept a `ResolveContext` directly.

Reply:
I am trying to avoid using `ResolveContext` where possible. This is some weird subset of `Configuration` and I think avoiding it will be in our favor (though I prefer it over using `Configuration`, since it does not expose as much)